Documentation/proposals/202304-silences.md

```diff
@@ -0,0 +1,112 @@
+# Silence CRD
+
+* Owners:
+  * [mcbenjemaa](https://github.com/mcbenjemaa)
+* Related Tickets:
+  * [#5452](https://github.com/prometheus-operator/prometheus-operator/issues/5452)
+  * [#2398](https://github.com/prometheus-operator/prometheus-operator/issues/2398)
+* Other docs:
+  * n/a
+
+This document describes the creation of `Silence` Custom Resource Definition that defines Alertmanager silences
+configurations in the Kubernetes way.
+
+# Why
+
+prometheus-operator doesn't have a way to automate the management of Alertmanager silences. Many users have either been using some internal scripts
+or an additional operator that does the job e.g. [silence-operator](https://github.com/giantswarm/silence-operator).
+
+* Users who are using CI/CD jobs to manage silences have reported that this is really cumbersome.
+* Users that use scripts are not in better shape obviously.
+* Users that use a standalone operator that exposes a Silence CRD are in a better situation, some folks have reported that they are using GitOps
+  to fully manage the life cycle of Silences Custom resources, Nevertheless, this comes with a drawback
+  as the team must be able to manage a new component in their stack.
+
+Additionally, having a new component in the stack and keeping it maintained is not always ideal (said the folks at [Giant Swarm](https://giantswarm.io) the owners of [silence-operator](https://github.com/giantswarm/silence-operator)),
+because that would require us to fully manage this component as of any managed app (monitoring, logging, alerting etc..).
+Having the Silence CRD as part of prometheus-operator provides users with a better experience: there is no need to install another operator or build a system on top of it to provide that functionality.
+
+By Adding support for `Silence` CRD in the prometheus-operator, this will make users more flexible in terms of choosing the tool
+to deploy their silences and will free others from managing a standalone component within the stack.
+
+## Pitfalls of the current solution
+
+Using Alertmanager API Directly comes with drawbacks:
+
+* Teams have to build an automation to add silences in a centralized manner
+* There is no input validation, which can lead to an invalid silence configuration
+* Teams need to manage expiration on their own
+* If a silence was deleted by accident, it will be permanent.
+
+# Goals
+
+* Provide a way for users to manage Alertmanager silences with the Kubernetes Way.
+* Make it easier to manage Alertmanager silences via centralised repo with e.g. ArgoCD, Flux...
+* Provide a way to manage silence expiration with a silence CR.
+
+## Audience
+
+* Users who serve Prometheus as a service and want to have an interface in defining silences exposed to developers.
+* Users who want to manage silences the same way as for services running within the Kubernetes cluster
+* Users who want a supported Kubernetes way of silences outside the Kubernetes cluster
```

Comment:
```suggestion
* Users who want to manage silences the same way as for services running within the Kubernetes cluster.
```

"Kubernetes way" sounds vague and ambiguous. And it is the same audience mentioned above right?